src/core/Workbench.tsx

```diff
@@ -1,267 +1,342 @@
-        }
-    }, []);
-    return [handleState, el, dispatch];
+  const [handleState, dispatch] = useReducer(handleWidthDragging, {
+    id,
+    invert,
+    dragging: false,
+  });
+  const el = useCallback((node) => {
+    if (node !== null) {
+      dispatch({ type: "widthknown", width: node.offsetWidth });
+    }
+  }, []);
+  return [handleState, el, dispatch];
 }
 
 function useHeightDraggableHandle(id: string, invert: boolean) {
-  const [handleState, dispatch] = useReducer(handleHeightDragging, { id, invert, dragging: false });
-  const el = useCallback(node => {
-      if (node !== null) {
-          dispatch({ type: "heightknown", height: node.offsetHeight });
-      }
+  const [handleState, dispatch] = useReducer(handleHeightDragging, {
+    id,
+    invert,
+    dragging: false,
+  });
+  const el = useCallback((node) => {
+    if (node !== null) {
+      dispatch({ type: "heightknown", height: node.offsetHeight });
+    }
   }, []);
   return [handleState, el, dispatch];
 }
 
 enum CssColor {
-    darkgray = "darkgray",
-    dimgray = "dimgray",
-    slategray = "slategray"
+  darkgray = "darkgray",
+  dimgray = "dimgray",
+  slategray = "slategray",
 }
 
 interface TextIconProps {
-    name: string;
-    color?: CssColor;
+  name: string;
+  color?: CssColor;
 }
 
-function TextIcon({name, color}: TextIconProps) {
-    return <span style={{color}} className="material-icons icon">{name}</span>;
-}
-
-function useToggle(initial: boolean): [boolean, () => void, () => void] {
-    const [value, setValue] = useState(initial);
-    return [
-        value,
-        () => { setValue(true) },
-        () => { setValue(false) }
-    ];
-}
-
-function useMouseUpAnywhere(onMouseUp) {
-    useEffect(() => {
-        document.addEventListener("mouseup", onMouseUp);
-        return () => document.removeEventListener("mouseup", onMouseUp);
-    }, []);
+function TextIcon({ name, color }: TextIconProps) {
+  return (
+    <span style={{ color }} className="material-icons icon">
+      {name}
+    </span>
+  );
 }
 
-function Tool({name, close, children}) {
-    const [minimizeChosen, chooseMinimize, unchooseMinimize] = useToggle(false);
-    useMouseUpAnywhere(unchooseMinimize);
-    const chosenClass = "minimize " + (minimizeChosen ? "chosen" : "");
-    return <>
-        <header className="tool-header">
-            <ul className="tool-header-main">
-                <li>{name}</li>
-            </ul>
-            <ul className="tool-header-controls">
-                <li className={chosenClass} onClick={close} onMouseDown={chooseMinimize} onMouseUp={unchooseMinimize}>
-                    <TextIcon color={CssColor.dimgray} name="minimize"/>
-                </li>
-            </ul>
-        </header>
-        <section className="tool">
-            {children}
-        </section>
-    </>;
+function Tool({ name, close, children }) {
+  const [minimizeChosen, _, setMinimize] = useToggle(false);
+  useMouseUpAnywhere(setMinimize(false));
```

Comment:
Small usage change here so that useToggle returns a toggle function so that it can be used without knowing the current value